File: android/src/main/kotlin/id/bni46/microatmsdk/data/ProgressMessage.kt
package id.bni46.microatmsdk.data

import androidx.annotation.Keep
import com.google.gson.Gson

@Keep
data class ProgressMessage(val message: String, val hasProgress: Boolean = true) {
    fun toJson(): String? {
        return Gson().toJson(this)
    }
}

File: lib/data/progress_message.dart
part of '../microatmsdk.dart';

class ProgressMessage {
  late String message;
  late bool hasProgress;

  ProgressMessage({
    this.message = "",
    this.hasProgress = false,
  });
  ProgressMessage.fromJson(Map<String, dynamic> json) {
    message = json['message'];
    hasProgress = json['hasProgress'];
  }
  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['message'] = message;
    data['hasProgress'] = hasProgress;
    return data;
  }
}
